### ISSUE TITLE
Looking for an easy way to mock a ConnectException

### ISSUE THREAD

**Issue by eduarguz** [Sammyjo20/Saloon#45]
Was looking for ways to mock not so common exceptions as `Guzzle/ConnectExceptions`

I think closures could work as in laravel HttpClient

https://laravel.com/docs/9.x/http-client#fake-callback

Also guzzle handlers allow these kind of mocked exceptions 

https://docs.guzzlephp.org/en/stable/testing.html#mock-handler

Any ideas?

**Comment by eduarguz** [comment]
Propossing something like:

``` 
new MockClient([
    fn() => new MockResponse(['name' => 'Sam'], 200),
    new MockResponse(['name' => 'Alex'], 200),
    new MockResponse(['error' => 'Server Unavailable'], 500),
]);

**Comment by Sammyjo20** [comment]
Thank you for mentioning this issue. I hadn't considered mocking exceptions but I have now created a PR for them #49 and here's how it will work:

```php
$mockClient = new MockClient([
    MockResponse::make(['name' => 'Sam'])->throw(new Exception('Custom Exception!')),
]);

// Or using Guzzle

$mockClient = new MockClient([
    MockResponse::make(['name' => 'Sam'])->throw(fn ($guzzleRequest) => new GuzzleException('Cannot connect!', $guzzleRequest)),
]);
```

It will be in the next release 🚀

**Comment by Sammyjo20** [comment]
Released in 0.11.0 / 0.12.0 (Laravel)
